Pi-hole Web v6.2

PromoFaux opened the pull request:
<!-- Release notes generated using configuration in .github/release.yml at development -->

## What's Changed
* Streamline actions, format and switch to Node.js 22 by @XhmikosR in https://github.com/pi-hole/web/pull/3357
* Remove remaining hard-coded /admin/ paths in the webinterface by @DL6ER in https://github.com/pi-hole/web/pull/3370
* Fix Local CNAME records records mistake by @yubiuser in https://github.com/pi-hole/web/pull/3367
* Fix concurrency issue about const apiUrl by @DL6ER in https://github.com/pi-hole/web/pull/3371
* groups-lists: remove unneeded HTML escape by @XhmikosR in https://github.com/pi-hole/web/pull/3379
* login.js: fix deprecated submit() usage by @XhmikosR in https://github.com/pi-hole/web/pull/3380
* Remove `x-dns-prefetch-control` meta tag by @XhmikosR in https://github.com/pi-hole/web/pull/3378
* group-domains: fix uncaught TypeError due to invalid URLs by @XhmikosR in https://github.com/pi-hole/web/pull/3383
* Replace remaining `i` element cases with `em` by @XhmikosR in https://github.com/pi-hole/web/pull/3381
* Revert the `defer` addition for now by @XhmikosR in https://github.com/pi-hole/web/pull/3382
* header: fix runtime error when query_string is null by @XhmikosR in https://github.com/pi-hole/web/pull/3395
* utils: fix typo in showAlert() by @XhmikosR in https://github.com/pi-hole/web/pull/3403
* queries: make the Advanced filtering button occupy all space by @XhmikosR in https://github.com/pi-hole/web/pull/3402
* Remove icheckbox related-code except for primary by @XhmikosR in https://github.com/pi-hole/web/pull/3387
* queries: add missing `display: none` by @XhmikosR in https://github.com/pi-hole/web/pull/3405
* Allow free input in upstream filter by @yubiuser in https://github.com/pi-hole/web/pull/3408
* Fix console warning when no password set by @PromoFaux in https://github.com/pi-hole/web/pull/3410
* Modernize JS and tighten things by @XhmikosR in https://github.com/pi-hole/web/pull/3388
* high-contrast.css: fix duplicate selector by @XhmikosR in https://github.com/pi-hole/web/pull/3411
* Remove unused CSS by @XhmikosR in https://github.com/pi-hole/web/pull/3420
* Add missing `display: none` for collapsed boxes by @XhmikosR in https://github.com/pi-hole/web/pull/3417
* Remove unused HTML elements by @XhmikosR in https://github.com/pi-hole/web/pull/3418
* Update chart.umd.js.map from npm by @XhmikosR in https://github.com/pi-hole/web/pull/3419
* settings-api: add missing end closing `p` tag by @XhmikosR in https://github.com/pi-hole/web/pull/3428
* settings-system: fix span end tag typo by @XhmikosR in https://github.com/pi-hole/web/pull/3427
* login: fix deprecated `submit()` function by @XhmikosR in https://github.com/pi-hole/web/pull/3425
* settings-api: fix deprecated `keyup()` function by @XhmikosR in https://github.com/pi-hole/web/pull/3424
* gravity: remove the unused `?go` code by @XhmikosR in https://github.com/pi-hole/web/pull/3426
* disabled and checked are boolean attributes by @XhmikosR in https://github.com/pi-hole/web/pull/3434
* queries: replace the deprecated `jQuery.parseJSON` with `JSON.parse` by @XhmikosR in https://github.com/pi-hole/web/pull/3435
* package.json: sort deps by name by @XhmikosR in https://github.com/pi-hole/web/pull/3437
* Use CODEOWNERS instead of deprecated dependbot/reviewers by @yubiuser in https://github.com/pi-hole/web/pull/3433
* Add 'Bug: Confirmed' to stale exempt list by @yubiuser in https://github.com/pi-hole/web/pull/3432
* Various minor tweaks to the LCARS theme by @rdwebdesign in https://github.com/pi-hole/web/pull/3446
* queries: fix box markup by @XhmikosR in https://github.com/pi-hole/web/pull/3445
* Use the same default interface used by FTL if none is set in pihole.toml by @rdwebdesign in https://github.com/pi-hole/web/pull/3436
* Use `font-weight: 600` by @XhmikosR in https://github.com/pi-hole/web/pull/3449
* Use mg.script_name to retrieve the scriptname by @rdwebdesign in https://github.com/pi-hole/web/pull/3447
* .user-header: use `font-weight: 600` by @XhmikosR in https://github.com/pi-hole/web/pull/3450
* settings-system.js: remove unused overlay code by @XhmikosR in https://github.com/pi-hole/web/pull/3455
* settings-advanced: remove unused code by @XhmikosR in https://github.com/pi-hole/web/pull/3454
* index.lp: remove unused classes by @XhmikosR in https://github.com/pi-hole/web/pull/3453
* scripts/js/settings-dns-records.js: fix api url base by @onlyJakob in https://github.com/pi-hole/web/pull/3456
* Add hint about wildcard search in advanced query filtering by @yubiuser in https://github.com/pi-hole/web/pull/3452
* index: make tables stripped by @XhmikosR in https://github.com/pi-hole/web/pull/3438
* Replace invalid hostname to prevent XSS by @yubiuser in https://github.com/pi-hole/web/pull/3401
* settings-system: fix typo by @XhmikosR in https://github.com/pi-hole/web/pull/3460
* sidebar: increase logo size to prevent a reflow by @XhmikosR in https://github.com/pi-hole/web/pull/3407
* queries: improve link to search page by @XhmikosR in https://github.com/pi-hole/web/pull/3422
* Hide top tables depending on privacy level by @yubiuser in https://github.com/pi-hole/web/pull/3362
* settings-dns-records: fix typo by @XhmikosR in https://github.com/pi-hole/web/pull/3459
* Fix deprecated `ready` function by @XhmikosR in https://github.com/pi-hole/web/pull/3461
* utils: remove the unused `objectToArray` util by @XhmikosR in https://github.com/pi-hole/web/pull/3462
* utils: improve readability for `validateIPv*` functions by @XhmikosR in https://github.com/pi-hole/web/pull/3423
* settings-api: fix wrong code by @XhmikosR in https://github.com/pi-hole/web/pull/3463
* Use `text()`/`textContent` when we don't need HTML by @XhmikosR in https://github.com/pi-hole/web/pull/3464
* footer: drop redundant addAdvancedInfo() call by @XhmikosR in https://github.com/pi-hole/web/pull/3467
* settings-system: remove unneeded span elements by @XhmikosR in https://github.com/pi-hole/web/pull/3468
* settings-dhcp: drop unused lookatme-text attribute by @XhmikosR in https://github.com/pi-hole/web/pull/3469
* settings-advanced.js: fix wrong `role` value by @XhmikosR in https://github.com/pi-hole/web/pull/3470
* settings-advanced: drop `small` element by @XhmikosR in https://github.com/pi-hole/web/pull/3472
* updateFtlInfo: remove unused code by @XhmikosR in https://github.com/pi-hole/web/pull/3474
* settings-level-expert: only toggle the expert elements by @XhmikosR in https://github.com/pi-hole/web/pull/3473
* Make use of the new format_path function to add the current page in body by @XhmikosR in https://github.com/pi-hole/web/pull/3390
* Use proper Object methods by @XhmikosR in https://github.com/pi-hole/web/pull/3441
* queries: minor grammar fix by @XhmikosR in https://github.com/pi-hole/web/pull/3477
* queries: change box to an info one by @XhmikosR in https://github.com/pi-hole/web/pull/3476
* Fix DataTables warning message in Network page by @rdwebdesign in https://github.com/pi-hole/web/pull/3475
* Fix broken themes by @rdwebdesign in https://github.com/pi-hole/web/pull/3479
* queries: drop non-existent elements code by @XhmikosR in https://github.com/pi-hole/web/pull/3478
* Reset `update-hint` text every time `updateVersionInfo()` function is executed by @rdwebdesign in https://github.com/pi-hole/web/pull/3480
* Use the same interface name used by FTL in DNS Settings page by @rdwebdesign in https://github.com/pi-hole/web/pull/3481
* messages: fix function call by @XhmikosR in https://github.com/pi-hole/web/pull/3482
* footer: rename label by @XhmikosR in https://github.com/pi-hole/web/pull/3486
* header_authenticated: change documentation icon to solid by @XhmikosR in https://github.com/pi-hole/web/pull/3488
* Remove horizontal line in hamburger menu by @yubiuser in https://github.com/pi-hole/web/pull/3491
* sidebar: move active class outside of class attribute conditional by @XhmikosR in https://github.com/pi-hole/web/pull/3492
* index: use sentence case for queries for consistency by @XhmikosR in https://github.com/pi-hole/web/pull/3493
* Add rel="noreferrer" to external hyperlinks by @casperklein in https://github.com/pi-hole/web/pull/3495
* settings-level-expert: hide them by default by @XhmikosR in https://github.com/pi-hole/web/pull/3487

## New Contributors
* @onlyJakob made their first contribution in https://github.com/pi-hole/web/pull/3456
* @casperklein made their first contribution in https://github.com/pi-hole/web/pull/3495

**Full Changelog**: https://github.com/pi-hole/web/compare/v6.1...v6.2

Commit:
Tighten xo rules

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
Manual xo fixes and suppressions

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
xo: enable strict mode

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
Fix package name to be valid

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
Streamline actions, format and switch to Node.js 22

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
Remove `x-dns-prefetch-control` meta tag

This is now included in FTL

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
groups-lists: remove unneeded HTML escape

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
Update Dockerfile

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

PromoFaux committed:
Sync master back into development (#3360)

yubiuser committed:
Streamline actions, format and switch to Node.js 22 (#3357)

yubiuser committed:
Hide top tables if no data is available

Signed-off-by: yubiuser <github@yubiuser.dev>

Commit:
login.js: fix deprecated submit() usage

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
Fix Local CNAME records records mistake

Signed-off-by: yubiuser <github@yubiuser.dev>

DL6ER committed:
Remove remaining hard-coded /admin/ paths in the webinterface

Signed-off-by: DL6ER <dl6er@dl6er.de>

DL6ER committed:
Always use document.body.dataset.apiurl instead of the intermediate const apiUrl

Signed-off-by: DL6ER <dl6er@dl6er.de>

XhmikosR committed:
Use proper Object methods

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
Use Object methods when possible

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

PromoFaux committed:
Remove remaining hard-coded /admin/ paths in the webinterface (#3370)

PromoFaux committed:
Fix Local CNAME records records mistake (#3367)

PromoFaux committed:
Fix concurrency issue about const apiUrl (#3371)

Commit:
Revert the `defer` addition for now

There are still cases we are getting TypeErrors. We should try again after grouping our assets together in all pages.

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
Replace remaining `i` element cases with `em`

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
group-domains: fix uncaught TypeError due to invalid URLs

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
Use for...of in more places

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
Fix deprecated `ready` function

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
sidebar: increase logo size to prevent a reflow

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

PromoFaux committed:
groups-lists: remove unneeded HTML escape (#3379)

PromoFaux committed:
login.js: fix deprecated submit() usage (#3380)

yubiuser committed:
Remove `x-dns-prefetch-control` meta tag (#3378)

PromoFaux committed:
group-domains: fix uncaught TypeError due to invalid URLs (#3383)

DL6ER committed:
Replace remaining `i` element cases with `em` (#3381)

Commit:
Remove icheckbox related-code except for primary

v6 doesn't support changing this like v5

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
Hide tables based on privacy level

Signed-off-by: yubiuser <github@yubiuser.dev>

Commit:
Make use of the new format_path function to add the current page in body

This will allow us to target specific pages more easily

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
Escape hostname to prevent XSS

Signed-off-by: yubiuser <github@yubiuser.dev>

dependabot[bot] committed:
Bump github/codeql-action from 3.28.13 to 3.28.15

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.28.13 to 3.28.15.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.28.13...v3.28.15)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 3.28.15
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
Bump github/codeql-action from 3.28.13 to 3.28.15 (#3393)

Commit:
header: fix runtime error when query_string is null

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
queries: make the Advanced filtering button occupy all space

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

dependabot[bot] committed:
Bump chart.js from 4.4.8 to 4.4.9

Bumps [chart.js](https://github.com/chartjs/Chart.js) from 4.4.8 to 4.4.9.
- [Release notes](https://github.com/chartjs/Chart.js/releases)
- [Commits](https://github.com/chartjs/Chart.js/compare/v4.4.8...v4.4.9)

---
updated-dependencies:
- dependency-name: chart.js
  dependency-version: 4.4.9
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
Update vendor files

Signed-off-by: yubiuser <github@yubiuser.dev>

Commit:
utils: fix typo in showAlert()

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
Bump chart.js from 4.4.8 to 4.4.9 (#3398)

Commit:
Remove unused CSS

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

dependabot[bot] committed:
Bump actions/setup-node from 4.3.0 to 4.4.0

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 4.3.0 to 4.4.0.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v4.3.0...v4.4.0)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-version: 4.4.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
Bump actions/setup-node from 4.3.0 to 4.4.0 (#3400)

DL6ER committed:
Revert the `defer` addition for now (#3382)

yubiuser committed:
header: fix runtime error when query_string is null (#3395)

yubiuser committed:
utils: fix typo in showAlert() (#3403)

yubiuser committed:
queries: make the Advanced filtering button occupy all space (#3402)

yubiuser committed:
Remove icheckbox related-code except for primary (#3387)

Commit:
queries: add missing `display: none`

Fixes the first expand not having animation


Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
Allow free input in upstream filter

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
queries: add missing `display: none` (#3405)

Commit:
Update chart.umd.js.map from npm

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
Allow free input in upstream filter (#3408)

PromoFaux committed:
use optional chaining for logout button event listener - ensures error not thrown when password is empty

Signed-off-by: Adam Warner <me@adamwarner.co.uk>

PromoFaux committed:
Fix console warning when no password set (#3410)

PromoFaux committed:
Modernize JS and tighten things (#3388)

Commit:
high-contrast.css: fix duplicate selector

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

rdwebdesign committed:
Remove duplicate selector from high-contrast.css (#3411)

dependabot[bot] committed:
Bump github/codeql-action from 3.28.15 to 3.28.16

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.28.15 to 3.28.16.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.28.15...v3.28.16)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 3.28.16
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
Bump github/codeql-action from 3.28.15 to 3.28.16 (#3414)

yubiuser committed:
Replace hostname string if it contains invalid characters

Signed-off-by: yubiuser <github@yubiuser.dev>

Commit:
Remove unused HTML elements

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
Add missing `display: none` for collapsed boxes

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
Remove deprecated <center>

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Remove unused CSS (#3420)

yubiuser committed:
Add missing `display: none` for collapsed boxes (#3417)

yubiuser committed:
Remove unused HTML elements (#3418)

yubiuser committed:
Update chart.umd.js.map from npm (#3419)

Commit:
queries: don't add a link to search if the domain is hidden

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
utils: improve readability for `validateIPv*` functions

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
settings-api: fix deprecated `keyup()` function

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
login: fix deprecated `submit()` function

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
gravity: remove the unused `?go` code

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
settings-system: fix span end tag typo

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
settings-api: add missing end closing `p` tag

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
settings-level-expert: only toggle the expert elements

Show the basic ones by default. Also, reduce jQuery usage in those functions while at it.

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
disabled and checked are boolean attributes

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
index: make tables stripped

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

dependabot[bot] committed:
Bump github/codeql-action from 3.28.16 to 3.28.17

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.28.16 to 3.28.17.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.28.16...v3.28.17)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 3.28.17
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
Bump github/codeql-action from 3.28.16 to 3.28.17 (#3429)

yubiuser committed:
settings-api: add missing end closing `p` tag (#3428)

yubiuser committed:
settings-system: fix span end tag typo (#3427)

yubiuser committed:
login: fix deprecated `submit()` function (#3425)

yubiuser committed:
settings-api: fix deprecated `keyup()` function (#3424)

yubiuser committed:
Add 'Bug: Confirmed' to stale exempt list

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Use CODEOWNERS instead of deprecated dependbot/reviewers

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
gravity: remove the unused `?go` code (#3426)

yubiuser committed:
disabled and checked are boolean attributes (#3434)

Commit:
queries: replace the deprecated `jQuery.parseJSON` with `JSON.parse`

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
queries: replace the deprecated `jQuery.parseJSON` with `JSON.parse` (#3435)

rdwebdesign committed:
Use the same default interface used by FTL if none is set in pihole.toml

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

Commit:
package.json: sort deps by name

This is the default when added/removing deps with npm

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
package.json: sort deps by name (#3437)

PromoFaux committed:
Use CODEOWNERS instead of deprecated dependbot/reviewers (#3433)

Commit:
queries: fix box markup

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

rdwebdesign committed:
Fix style for dropdown-menu (used on Group Management pages)

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Adjust buttons styles

- change "Apply" button (Group management pages)
- adjust teleporter buttons colors and padding
- adjust small buttons
- use "success" color for Local DNS buttons (all themes)

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

yubiuser commented:
Despite including many commits, they are mainly internal fixes/cleanups/modernization.  I think this would mainly be a patch release... 6.1.1.

rdwebdesign committed:
Use better alignment for tree elements in Tools-Interfaces page

Fixes #3303

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Other small LCARS improvements

- adjust status panel icons contrast using better colors
- increase general text contrast a little

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Use mg.script_name to retrieve the scriptname

This value is independent from webhome or prefix

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

yubiuser committed:
Add 'Bug: Confirmed' to stale exempt list (#3432)

yubiuser committed:
fix tests

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Various minor tweaks to the LCARS theme (#3446)

yubiuser committed:
queries: fix box markup (#3445)

rdwebdesign committed:
Use the same default interface used by FTL if none is set in pihole.toml (#3436)

XhmikosR committed:
Use `font-weight: 600`

We don't have 500 in the current source-sans-pro version

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
.user-header: use `font-weight: 600`

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
Use `font-weight: 600` (#3449)

rdwebdesign committed:
Use mg.script_name to retrieve the scriptname (#3447)

yubiuser committed:
.user-header: use `font-weight: 600` (#3450)

yubiuser committed:
Add hint about wildcard search in advanced query filtering

Signed-off-by: yubiuser <github@yubiuser.dev>

XhmikosR committed:
queries.js: link to CNAME if available

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
index.lp: remove unused classes

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
settings-advanced: remove unused code

There's no such button with id=save

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
settings-system.js: remove unused overlay code

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
settings-system.js: remove unused overlay code (#3455)

yubiuser committed:
settings-advanced: remove unused code (#3454)

yubiuser committed:
index.lp: remove unused classes (#3453)

onlyJakob committed:
scripts/js/settings-dns-records.js: fix api url base

Signed-off-by: Jakob Lorenz <onlyJak0b@mailbox.org>

DL6ER committed:
scripts/js/settings-dns-records.js: fix api url base (#3456)

yubiuser committed:
Add hint about wildcard search in advanced query filtering (#3452)

DL6ER committed:
index: make tables stripped (#3438)

DL6ER committed:
Replace invalid hostname to prevent XSS (#3401)

XhmikosR committed:
settings-dns-records: fix typo

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
settings-system: fix typo

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
settings-system: fix typo (#3460)

XhmikosR committed:
utils: remove the unused `objectToArray` util

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

PromoFaux committed:
sidebar: increase logo size to prevent a reflow (#3407)

XhmikosR committed:
settings-api: fix wrong code

`ids` is an array so `id` is the value when using a for...of loop.

Also, switch to Number.isInteger since we expect the IDs to be numbers

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

PromoFaux committed:
queries: improve link to search page (#3422)

PromoFaux committed:
Hide top tables depending on privacy level (#3362)

yubiuser committed:
settings-dns-records: fix typo (#3459)

PromoFaux committed:
Fix deprecated `ready` function (#3461)

PromoFaux committed:
utils: remove the unused `objectToArray` util (#3462)

PromoFaux committed:
utils: improve readability for `validateIPv*` functions (#3423)

XhmikosR committed:
network: use `textContent` when we don't need HTML

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
footer: use `text()` when we don't need HTML

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
charts.j: use `textContent` when we don't need HTML

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

dependabot[bot] committed:
Bump github/codeql-action from 3.28.17 to 3.28.18

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.28.17 to 3.28.18.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.28.17...v3.28.18)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 3.28.18
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
Bump github/codeql-action from 3.28.17 to 3.28.18 (#3465)

DL6ER committed:
settings-api: fix wrong code (#3463)

XhmikosR committed:
footer: drop redundant addAdvancedInfo() call

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
Use `text()`/`textContent` when we don't need HTML (#3464)

XhmikosR committed:
settings-system: remove unneeded span elements

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
footer: drop redundant addAdvancedInfo() call (#3467)

XhmikosR committed:
settings-system: drop unused lookatme-text attribute

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
settings-dhcp: drop unused lookatme-text attribute

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
settings-advanced.js: fix wrong `role` value

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
settings-system: remove unneeded span elements (#3468)

yubiuser committed:
settings-dhcp: drop unused lookatme-text attribute (#3469)

yubiuser committed:
settings-advanced.js: fix wrong `role` value (#3470)

XhmikosR committed:
settings-advanced: drop `small` element

Use `.small` on the wrapper element

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
updateFtlInfo: remove unused code

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

PromoFaux committed:
settings-advanced: drop `small` element (#3472)

PromoFaux committed:
updateFtlInfo: remove unused code (#3474)

PromoFaux committed:
settings-level-expert: only toggle the expert elements (#3473)

DL6ER committed:
Make use of the new format_path function to add the current page in body (#3390)

PromoFaux committed:
Use proper Object methods (#3441)

rdwebdesign committed:
Fix DataTables warning message

Use `null` instead of an empty string to define columns initially empty

Fix #3471

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

XhmikosR committed:
queries: change box to an info one

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
queries: minor grammar fix

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
queries: minor grammar fix (#3477)

yubiuser committed:
queries: change box to an info one (#3476)

yubiuser committed:
Fix DataTables warning message in Network page (#3475)

XhmikosR committed:
queries: drop non-existent elements code

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

rdwebdesign committed:
Fix "Advanced Filtering" colors in High Contrast Dark theme

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Fix `.box-title` color (LCARS theme)

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Fix "Advanced Filtering" colors in Dark theme 

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Reset `update-hint` text every time the function is executed

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Fix broken themes (#3479)

rdwebdesign committed:
Use the same interface name used by FTL

If `dns.interface` is empty in pihole.toml we use the first interface, just like FTL.

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

yubiuser committed:
queries: drop non-existent elements code (#3478)

yubiuser committed:
Reset `update-hint` text every time `updateVersionInfo()` function is executed (#3480)

XhmikosR committed:
messages: fix function call

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

rdwebdesign committed:
Use the same interface name used by FTL in DNS Settings page (#3481)

yubiuser committed:
messages: fix function call (#3482)

XhmikosR committed:
footer: rename label

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
settings-level-expert: hide them by default

Matches the previous behavior before 8556b65

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

XhmikosR committed:
header_authenticated: change documentation icon to solid

It's now consistent with the rest of the icons

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
footer: rename label (#3486)

yubiuser committed:
header_authenticated: change documentation icon to solid (#3488)

yubiuser committed:
Remove horizontal line in hamburger menu

Signed-off-by: yubiuser <github@yubiuser.dev>

rdwebdesign committed:
Remove horizontal line in hamburger menu (#3491)

XhmikosR committed:
sidebar: move active class outside of class attribute conditional

Gets rid of empty class attributes when not needed

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
sidebar: move active class outside of class attribute conditional (#3492)

XhmikosR committed:
index: use sentence case for queries for consistency

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
index: use sentence case for queries for consistency (#3493)

casperklein committed:
Add rel="noreferrer" to external hyperlinks

Signed-off-by: casperklein <casperklein@users.noreply.github.com>

yubiuser committed:
Add rel="noreferrer" to external hyperlinks (#3495)

PromoFaux committed:
settings-level-expert: hide them by default (#3487)
